src/pages/signup.tsx

```diff
@@ -19,21 +18,18 @@ const BlobCircle = () => {
   );
 };
 
-const getTimezone = (): string => {
-  return Intl.DateTimeFormat().resolvedOptions().timeZone;
-};
-
 const SignUpPage: Page = () => {
-  const { signUpWithEmail, signInWithGoogle, signOut } = useAuth();
-  const [createUser] = useCreateUserMutation();
-  const [displayError, setDisplayError] = useState("");
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
-  const [firstName, setFirstName] = useState("");
-  const [lastName, setLastName] = useState("");
-  // const [tocChecked, setTocChecked] = useState(false);
+  const authorizationLevel = useAuthorizationLevel();
   const router = useRouter();
 
+  if (
+    authorizationLevel !== AuthorizationLevel.Unverified &&
+    authorizationLevel !== AuthorizationLevel.Unauthenticated
+  ) {
+    router.push("/");
+    return <div>Logged in. Redirecting...</div>;
```

Comment:
Maybe return a fragment or some other loading screen (text is jank)

(Same for login)